src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+        return "http://localhost:" + port + "/users";
+    }
+
+    private String getBaseProductUrl() {
+        return "http://localhost:" + port + "/products";
+    }
+
+    private String getProductUrl(UUID id) {
+        return getBaseUrl() + "/products/" + id;
+    }
+
+    @Autowired
+    TestRestTemplate testRestTemplate;
+    @Autowired
+    UserRepository userRepository;
+    @Autowired
+    ProductRepository productRepository;
+    @Autowired
+    ResourceRepository resourceRepository;
+    @Autowired
+    ResourceInUserRepository resourceInUserRepository;
+    @Autowired
+    ResourceInProductRepository resourceInProductRepository;
+
+    User user;
+    Gemstone gemstone;
+    ResourceInUserRequestDto resourceInUserRequestDto;
+    ResourcesInUserResponseDto resourcesInUser;
+    ProductRequestDto productRequestDto;
+    ProductResponseDto productResponseDto;
+
+    @BeforeEach
+    void setUp() {
+        userRepository.deleteAll();
+        productRepository.deleteAll();
+        resourceRepository.deleteAll();
+        resourceInUserRepository.deleteAll();
+        resourceInProductRepository.deleteAll();
+
+        user = getUser();
+        gemstone = getGemstone();
+        resourceInUserRequestDto = getResourceInUserRequestDto(user, Objects.requireNonNull(gemstone));
+        resourcesInUser = getResourcesInUserResponseDto(resourceInUserRequestDto);
+        productRequestDto = getProductRequestDto(Objects.requireNonNull(resourcesInUser), user);
+    }
+
+    @Test
+    void createProductSuccessfully() {
+
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.postForEntity(getBaseProductUrl(), productRequestDto, ProductResponseDto.class);
+        ProductResponseDto productResponseDto = response.getBody();
+
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+
+        assertEquals(productRequestDto.getName(), Objects.requireNonNull(productResponseDto).getName());
+        assertEquals(productRequestDto.getResourcesContent().get(0).getId(), productResponseDto.getResourcesContent().get(0).getResource().getId());
+    }
+
+    @Test
+    void getProductSuccessfully() {
+
+        productResponseDto = getProductResponseDto(productRequestDto);
+
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.getForEntity(getProductUrl(Objects.requireNonNull(productResponseDto).getId()), ProductResponseDto.class);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    void getAllProductsSuccessfully() {
+        productResponseDto = getProductResponseDto(productRequestDto);
+
+        ResponseEntity<List<ProductResponseDto>> response =
+                this.testRestTemplate.exchange(
+                        getBaseProductUrl(), HttpMethod.GET, null, new ParameterizedTypeReference<>() {
+                        });
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
```

Comment:
Same here, we should validate that there is a product created in the db with the properties we have put in the request.